docs/migration-service.md

```diff
@@ -122,96 +200,242 @@ flowchart TD
+> **Hard Launch Extension (Planned):** `BackupServiceInfo` will be extended with `attestation: Option<Attestation>` and `registered_at: Timestamp` fields for TEE attestation verification and expiration tracking.
 
+**Hard Launch Extensions:**
 
-#### Cryptography
-A pair-wise key establishment scheme can be leveraged to establish a symmetric key. A suitable implementation can be found in [libsodium](https://doc.libsodium.org/) and its rust wrapper [libsodium-rs](https://docs.rs/crate/libsodium-rs/latest). Specifically, the **key exchange protocol based on X25519** would be a good fit for the given set-up (c.f. [libsodium](https://doc.libsodium.org/key_exchange) and [libsodium-rs](https://docs.rs/libsodium-rs/latest/libsodium_rs/crypto_kx/index.html)).
+For hard launch, the backup service must provide TEE attestation similar to MPC nodes. The contract will need to be extended to support attestation verification for backup services.
 
-The key establishment scheme requires that the node as well as the backup service have *mutually authenticated Curve25519 public keys*. The NEAR blockchain can be leveraged for this, more specifically, for each backup generation or recovery, the node and the backup service generate ephemeral keys on the `Curve25519` and publish them on the MPC smart contract. The node and the backup service can then each run a key generation protocol using their private key and the public key of the other party. 
+**Planned Implementation Approach:**
 
-_Note: The curious reader might ask why this protocol does not simply use the NEAR public/private key pairs associated to the MPC node and the backup service. The reason for this is is twofold:_
-- _those keys are meant for signature generation._
-- _While it is true that Curve25519 used in X25519 and edwards25519 used by the NEAR blockchain are [birationally equivalent](https://crypto.stackexchange.com/questions/43013/what-does-birational-equivalence-mean-in-a-cryptographic-context), so one could theoretically convert the NEAR account keys and use them for `X25519`, it is generally advised to use one key per application. This also allows us to use ephemeral keys, as opposed to static keys for the encryption. Which is desirable._
+The backup service identification remains the same as soft launch: `AccountId` maps to `BackupServiceInfo`. For hard launch, the `BackupServiceInfo` struct would be extended to include attestation:
 
+```Rust
+/// Backup service authentication information (PLANNED extension for hard launch)
+pub struct BackupServiceInfo {
+    /// Ed25519 public key for mutual TLS authentication
+    pub public_key: Ed25519PublicKey,
+    
+    /// TEE attestation proving backup service runs approved code (hard launch only)
+    pub attestation: Option<Attestation>,
+    
+    /// Timestamp when backup service registered (for attestation expiration tracking)
+    pub registered_at: Option<Timestamp>,
```

Comment:
I think we can make them non-optional, but we support mocked attestations in the beginning, similar to how we handle the node. This way, we won't need specific code migration logic when we go from soft to hard launch. Wdyt?